docs/spec/v1alpha1/gitrepositories.md

```diff
@@ -0,0 +1,212 @@
+# Git Repositories
+
+The `GitReposiory` API defines a source for artifacts coming from Git. 
+
+## Specification
+
+Git repository spec:
+
+```go
+// GitRepositorySpec defines the desired state of GitRepository
+type GitRepositorySpec struct {
+	// +kubebuilder:validation:Pattern="^(http|https|ssh)://"
+
+	// The repository URL, can be a HTTP or SSH address.
+	Url string `json:"url"`
+
+	// The secret name containing the Git credentials
+	// +optional
+	SecretRef *v1.LocalObjectReference `json:"secretRef,omitempty"`
+	
+	// The git reference to checkout and monitor for changes.
+	// +optional
+	Reference *GitRepositoryRef `json:"ref,omitempty"`
+
+	// The interval at which to check for repository updates.
+	Interval metav1.Duration `json:"interval"`
+}
+
+// GitRepositoryRef defines the git ref used for pull and checkout operations
+type GitRepositoryRef struct {
```

Comment:
Is this is union (you supply one of these) or can you supply more than one value? What's the meaning of supplying more than one value?

Reply:
The precedence is described on each field: tag semver > tag > branch

Reply:
Sure. But why would you supply any field other than the one with the highest precedence? E.g., if I want to specify a tag, why give the branch? If I want to specify a semver range, why supply a branch or tag?

Reply:
We can enforce a single field to be supplied with `oneOf` validation. I just need to figure out the right kubebuilder annotations for this.

Reply:
OK, I see -- the intention is for only one field to have effect, and the precedence rule is a way of getting that. (But others exist) :+1:

Reply:
> But why would you supply any field other than the one with the highest precedence?

This wasn't the intention, only one field should be supplied, the precedence is there in case we can't use `oneOf` validation with kubebuilder.

Reply:
I've added examples for each supported combination.